src/routes/transactions/mappers/common/bridge-transaction.mapper.ts

```diff
@@ -0,0 +1,187 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { LiFiDecoder } from '@/domain/bridge/contracts/decoders/lifi-decoder.helper';
+import {
+  BridgeTransactionInfo,
+  SwapAndBridgeTransactionInfo,
+  SwapTransactionInfo,
+} from '@/routes/transactions/entities/bridge/bridge-info.entity';
+import { ITokenRepository } from '@/domain/tokens/token.repository.interface';
+import { TokenInfo } from '@/routes/transactions/entities/swaps/token-info.entity';
+import { AddressInfoHelper } from '@/routes/common/address-info/address-info.helper';
+import { IBridgeRepository } from '@/domain/bridge/bridge.repository.interface';
+import { BridgeStatus } from '@/domain/bridge/entities/bridge-status.entity';
+import {
+  BridgeName,
+  isBridgeName,
+} from '@/domain/bridge/entities/bridge-name.entity';
+
+@Injectable()
+export class BridgeTransactionMapper {
+  constructor(
+    private readonly liFiDecoder: LiFiDecoder,
+    @Inject(ITokenRepository)
+    private readonly tokenRepository: ITokenRepository,
+    private readonly addressInfoHelper: AddressInfoHelper,
+    @Inject(IBridgeRepository)
+    private readonly bridgeRepository: IBridgeRepository,
+  ) {}
+
+  // TODO:
+  public mapBridge(data: `0x${string}`): BridgeTransactionInfo | null {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const decoded = this.liFiDecoder.decodeBridgeAndMaybeSwap(data);
+
+    return new BridgeTransactionInfo();
+  }
+
+  // TODO:
+  public mapSwap(data: `0x${string}`): SwapTransactionInfo {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const decoded = this.liFiDecoder.decodeSwap(data);
+
+    return new SwapTransactionInfo();
+  }
+
+  public async mapSwapAndBridge(args: {
+    chainId: string;
+    data: `0x${string}`;
+    executionDate: Date | null;
+    safeAddress: `0x${string}`;
+  }): Promise<SwapAndBridgeTransactionInfo | null> {
+    const decoded = this.liFiDecoder.decodeBridgeAndMaybeSwap(args.data);
+
+    const [fromToken, toToken, recipient, info] = await Promise.all([
+      this.tokenRepository.getToken({
+        address: decoded.fromToken,
+        chainId: args.chainId,
+      }),
+      this.tokenRepository.getToken({
+        address: decoded.toToken,
+        chainId: args.chainId,
+      }),
+      this.addressInfoHelper.getOrDefault(args.chainId, decoded.toAddress, [
+        'TOKEN',
+        'CONTRACT',
+      ]),
+      args.executionDate
+        ? this.getHistoricalInfo({
+            ...decoded,
+            chainId: args.chainId,
+          })
+        : this.getQueuedInfo({
+            fromChain: args.chainId,
+            toChain: args.chainId,
```

Comment:
I think we need to use the destination chain from the decoded transaction data here.
```suggestion
            toChain: decoded.toChain,
```